src/dataframerow/dataframerow.jl

```diff
@@ -203,3 +203,39 @@ function Base.isless(r1::DataFrameRow, r2::DataFrameRow)
     end
     return false
 end
+
+function DataFrame(dfr::DataFrameRow)
+    row, cols = parentindices(dfr)
+    parent(dfr)[row:row, cols]
+end
+
+pushhelper!(x, r) = push!(x, x[r])
```

Comment:
I don't understand the advantage of this helper. :-)

Reply:
Have you checked it really makes a difference? Without `@noinline` I'd expect it to be inlined since it's short, which would defeat its purpose. Also a comment would be useful.

Reply:
I would say that the difference is relevant.

Here are the benchmarks:

my approach using a barrier function:
```
julia> df = DataFrame(rand(100, 100));

julia> dfr = df[10, :];

julia> @benchmark push!($df, $dfr)
BenchmarkTools.Trial:
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     2.113 μs (0.00% GC)
  median time:      2.228 μs (0.00% GC)
  mean time:        2.676 μs (0.00% GC)
  maximum time:     48.664 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     9
```

the approach without a barrier function:
```
julia> df = DataFrame(rand(100, 100));

julia> dfr = df[10, :];

julia> @benchmark push!($df, $dfr)
BenchmarkTools.Trial:
  memory estimate:  1.56 KiB
  allocs estimate:  100
  --------------
  minimum time:     3.791 μs (0.00% GC)
  median time:      3.920 μs (0.00% GC)
  mean time:        4.704 μs (0.59% GC)
  maximum time:     61.044 μs (91.37% GC)
  --------------
  samples:          10000
  evals/sample:     8
```

Reply:
I have added a comment in the code.

Reply:
OK. How about adding `@noinline` just to be sure, in case the compiler changes its strategy?

Reply:
Added. These compiler-level things with Julia are sometimes tricky 😄.